app/code/Magento/ConfigurableProduct/Model/Product/VariationHandler.php

```diff
@@ -239,6 +239,10 @@ public function duplicateImagesForVariations($productsData)
 
             foreach ($simpleProductData['media_gallery']['images'] as $imageId => $image) {
                 $image['variation_id'] = $variationId;
+                if (isset($imagesForCopy[$imageId][0])) {
+                    // skip duplicate image for first product
+                    unset($imagesForCopy[$imageId][0]);
+                }
```

Comment:
Seems like this change isn't related to this PR and causing functional test failures. Could you revert it?
```suggestion
```

Reply:
Hi! @ihor-sviziev I am not actually a member of the team nor involved in this PR.

Reply:
@aredridel oh sorry, just mentioned wrong account. I fixed already my comment